pkg/commands/models/branch.go

```diff
@@ -32,6 +35,11 @@ type Branch struct {
 	Subject string
 	// commit hash
 	CommitHash string
+
+	// How far we have fallen behind our base branch. 0 means either not
+	// determined yet, or up to date with base branch. (We don't need to
+	// distinguish the two, as we don't draw anything in both cases.)
+	BehindBaseBranch atomic.Int32
```

Comment:
It is read and written concurrently, so a plain int would be wrong. Granted, on most modern processor architectures a plain int would probably work fine in practice, but it's better not to rely on that. And running with the `-race` option would very likely flag it as an error.

I also think that using atomic values can probably help solve many of the other concurrency issues we have. I talked a few times about my plan to restructure our code so that all changes to the model are only done on the main thread, but I no longer think that's the best approach; using atomic values for the model slices should make it possible to update them safely from background threads without having to do use a mutex. But that's for another day...

Reply:
Makes sense. I still feel that having a concurrency model with fewer mutexes/atomics would be preferable to one where it's more fine-grained, but let's not get bogged down in that now.